### PR TITLE
Display for BMI in clinic and vitals

### PR DESCRIPTION
**Pull request by ynurmahomed**
Transformed visit header into a component so it can be reused across modules. 
Refactored the services used to load visit header information.

### CODE REVIEW

**Comment by ynurmahomed** [comment]
[7.3.3. BMI: Develop the calculation and display for BMI in clinic and vitals pages](https://trello.com/c/yIMkRw04/45-733-bmi-develop-the-calculation-and-display-for-bmi-in-clinic-and-vitals-pages)

**Comment by valvijo** [comment]
Hi @ynurmahomed I cannot merge this unless conflicts are resolved

**Comment by ynurmahomed** [comment]
@valvijo conflicts resolved. I suggest that these commits be squashed and merged.

**Comment by valvijo** [comment]
The problem here is that BMI obs `(ID: 1342, UUID: e1da52ba-1d5f-11e0-b929-000c29ad1d07)` is not being generated and inserted in the database when the vitals are inserted. I see two options to solve this:
(1) Calculate BMI and create a database record for the `obs`. The display gets the stored `obs` value. This is  how RDE works.
(2) Don't create an `obs` record for the BMI but calculate it for display. The good about this is that we don't have to make sure that BMI obs is updated in case height and/or weight are updated

**Comment by ynurmahomed** [comment]
I will go with (2), don't want to mess with form stuff. :grin:

**Comment by ynurmahomed** [comment]
@valvijo

File: app/visit/visitService.js [hunk 1]
```diff
@@ -109,6 +113,22 @@
...
 
+
+    function getPatientLastBMI(patient) {
+      var getHeightCM = observationsService.getLastPatientObs(patient, {uuid: HEIGHT_CM}, 'custom:(uuid,display,value,obsDatetime)');
+      var getWeightKg = observationsService.getLastPatientObs(patient, {uuid: WEIGHT_KG}, 'custom:(uuid,display,value)');
+      return $q.all([getHeightCM, getWeightKg])
+        .then(function (result) {
+          var heightCMObs = result[0];
+          var weightKgObs = result[1];
+          var heightM = heightCMObs.value / 100;
```
Comment (valvijo) [inline app/visit/visitService.js]: No need to divide by 100, height is provided in meters in the POC UI. The only problem is that RDE allows either meters and centimeters in the input. Can this check whether the value is `m` or `cm` and calculate accordingly?

File: app/visit/components/visitHeader.html [hunk 2]
```diff
@@ -45,7 +45,7 @@
         <b>{{'VITALS_LAST_BMI'| translate}}: </b>
       </div>
       <div class="col-sm-9 col-md-10" ng-if="vm.lastBmi.value">
-        <span class="label label-default">{{vm.lastBmi.value}}</span>
+        <span class="label label-default">{{vm.lastBmi.value | number:2}}</span>
```
Comment (valvijo) [inline app/visit/components/visitHeader.html]: The BMI standard is 1 decimal place `number:1`

File: app/visit/visitService.js (hunk 1 above)
Comment (ynurmahomed) [reply app/visit/visitService.js]: Looks like [HEIGHT (CM)](http://dev.esaude.org:8080/openmrs/dictionary/concept.htm?conceptId=5090) expects the values to be in ```cm```, should we really be saving them in ```m```? How to tell which one is being used? There is also this issue https://github.com/esaude/esaude-emr-poc/issues/361.

Comment (valvijo) [reply app/visit/visitService.js]: I know but a requirements from one of the previous system reviews said that height should be on `m` in order to align with the paper forms. I don't really know why it's `cm` in RDE! Anyway we have to deal with it :unamused:

**Comment by ynurmahomed** [comment]
@valvijo

**Pull request merged**